src/main/java/pingis/services/UserService.java

```diff
@@ -101,4 +105,17 @@ public User findByName(String name) {
   public User getCurrentUser() {
     return findByName(SecurityContextHolder.getContext().getAuthentication().getName());
   }
+
+  public int getLevel(int score) {
+    for (int i = 1; i <= numberOfLevels; i++) {
```

Comment:
Couldn't this be computed with logarithms instead, e.g. `level = log(points, base) - log(pointsForFirstLevel, base)`?

Reply:
Also needs more tests.